packages/material-ui/src/StepConnector/StepConnector.js

```diff
@@ -38,10 +38,25 @@ export const styles = theme => ({
     borderLeftWidth: 1,
     minHeight: 24,
   },
+  /* Styles applied to the line element if `active={true}`. */
+  lineActive: {},
```

Comment:
What do you think of moving the state variant to the root element so people can write advanced logic?

Reply:
What do you mean by root element? The `div` element on the `StepConnector` component? So basically move these

```
[classes.lineActive]: active,
[classes.lineCompleted]: completed,
[classes.lineDisabled]: disabled,
```

to `className` instead of the `lineClassName`? So they are applied to the root element?

Reply:
Yes, what do you think about it.

Reply:
I tried that! However, the following doesn't work anymore

```jsx
lineActive: {
  borderColor: theme.palette.secondary.main,
},
lineCompleted: {
  borderColor: theme.palette.primary.main,
},

<StepConnector
  classes={{ lineActive: classes.lineActive, lineCompleted: classes.lineCompleted }}
/>
```

because the `div` element doesn't have a `borderColor`. How can we apply a `borderColor` in this case with a class on the `div` element instead of the `span` element? How can we style the line with a class on the `div` element? Maybe I am missing something.

Reply:
Ok, there is a way to accomplish that! I updated the PR!